test/Microsoft.AspNetCore.Http.Connections.Tests/WebSocketsTests.cs

```diff
@@ -24,10 +24,18 @@ public WebSocketsTests(ITestOutputHelper output)
         {
         }
 
+        // For unknown reasons using the built-in WebSocketMessageType as a parameter
+        // causes "Non-serializable data" warnings in xunit
+        public enum TestWebSocketMessageType
+        {
+            Binary = 1,
+            Text = 0
```

Comment:
Might be worth filing a bug, but I don't know exactly where it would go. I think maybe it's xunit here? `WebSocketMessageType` doesn't look any different from this enum really: https://source.dot.net/#System.Net.WebSockets/System/Net/WebSockets/WebSocketMessageType.cs,17aa513fb8c7e9fc

Reply:
Sounds like it's a problem on desktop framework when the enum is in the GAC. https://twitter.com/anurse/status/980825964757176320

Reply:
https://github.com/xunit/xunit/blob/4d2679934462bc2bc459f340d8bcfce624ce1512/test/test.xunit.execution/Common/XunitSerializationInfoTests.cs#L200-L210

Reply:
And not just enums, anything from the GAC (that we don't already special case): https://github.com/xunit/xunit/blob/4d2679934462bc2bc459f340d8bcfce624ce1512/test/test.xunit.execution/Common/XunitSerializationInfoTests.cs#L233-L245

Reply:
Another reason to hate the GAC